src/Microsoft.AspNetCore.Server.Testing/Deployers/NginxDeployer.cs

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Net.Http;
+using Microsoft.AspNetCore.Server.Testing.Common;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Server.Testing
+{
+    /// <summary>
+    /// Deployer for Kestrel on Nginx.
+    /// </summary>
+    public class NginxDeployer : SelfHostDeployer
+    {
+        private string configFile;
+        private readonly int waitTime = (int)TimeSpan.FromSeconds(2).TotalMilliseconds;
```

Comment:
On my machine the commands all return instantly, can't test on CI until this actually goes through